modules/edge-voting/src/voting.rs

```diff
@@ -248,10 +219,40 @@ impl<T: Trait> Module<T> {
 		Ok(())
 	}
 
+	pub fn reveal_unsigned(sender: T::AccountId, vote_id: u64, vote: [u8; 32], secret: Option<[u8; 32]>) -> Result {
+		let mut record = <VoteRecords<T>>::get(vote_id).ok_or("Vote record does not exist")?;
+		ensure!(record.data.stage == VoteStage::Voting, "Vote is not in voting stage");
+		// Check vote is for a valid outcome
+		ensure!(record.outcomes.iter().any(|o| o == &vote), "Vote type must be binary");
+		// TODO: Allow changing of votes
+		ensure!(!record.reveals.iter().any(|c| &c.0 == &sender), "Duplicate votes are not allowed");
```

Comment:
We may want to maintain a storage HashMap that tracks the existence of voters reveals so that we don't need to search through this as well.

Reply:
Something akin to `map: (VoteId, T::AccountId) => bool`